Update SYNC_UPSTREAM.md

aaronenyeshi opened the pull request:
I've updated the SYNC_UPSTREAM.md document for HCC Merge Maintenance. It has been changed so that users who do not have write access to RadeonOpenCompute/hcc can perform HCC Merge on their own forks and branches.

However, I did not modify the repo or manifest descriptions.

aaronenyeshi committed:
Update SYNC_UPSTREAM.md

I've updated the SYNC_UPSTREAM.md document for HCC Merge Maintenance. It has been changed so that users who do not have write access to RadeonOpenCompute/hcc can perform HCC Merge on their own forks and branches.

However, I did not modify the repo or manifest descriptions.

whchung commented on SYNC_UPSTREAM.md:
```diff
@@ -205,58 +202,69 @@ bin/hcc `bin/hcc-config --build --cxxflags --ldflags` \
 - `cd clang`
-- `git checkout clang_tot_upgrade`
-- `git push`
+- `git push clang_fork merge_20170626:merge_20170626`
```
@aaronenyeshi there are date-specific information here. could you make it more general?

whchung commented on SYNC_UPSTREAM.md:
```diff
@@ -205,58 +202,69 @@ bin/hcc `bin/hcc-config --build --cxxflags --ldflags` \
 - `cd lld`
-- `git checkout amd-hcc`
-- `git push`
+- `git push lld_fork merge_20170626:merge_20170626`
```
@aaronenyeshi same here.

whchung commented on SYNC_UPSTREAM.md:
```diff
@@ -205,58 +202,69 @@ bin/hcc `bin/hcc-config --build --cxxflags --ldflags` \
 - `cd compiler`
-- `git checkout amd-hcc`
-- `git push`
+- `git push llvm_fork merge_20170626:merge_20170626`
```
@aaronenyeshi same here.

whchung commented on SYNC_UPSTREAM.md:
```diff
@@ -205,58 +202,69 @@ bin/hcc `bin/hcc-config --build --cxxflags --ldflags` \
 - `cd compiler-rt`
-- `git checkout amd-hcc`
-- `git push`
+- `git push compiler-rt_fork merge_20170626:merge_20170626`
```
@aaronenyeshi same here.

whchung commented on SYNC_UPSTREAM.md:
```diff
@@ -205,58 +202,69 @@ bin/hcc `bin/hcc-config --build --cxxflags --ldflags` \
-- `cd rocdl`
-- `git checkout remove-promote-change-addr-space`
-- `git push`
+*** Wait until all Pull Requests are approved and merged. ***
```
@aaronenyeshi since now the whole process can't be completed without the PRs be merged. perhaps you may want to add another paragraph or 2 to illustrate the interaction between one who initates the PRs, and the one who reviews them?

whchung reviewed (changes requested):
@aaronenyeshi thanks for contributing this PR. i've read the proposed changes and raised a few places to revise. could you help revise the document to make it even better? thanks.

whchung commented on SYNC_UPSTREAM.md:
```diff
@@ -8,12 +8,12 @@ Upstream Clang, LLVM, and LLD all sit in different git repositories and they
+and may contain patches not yet upstreamed. amd-common LLVM / LLD / Clang is
 automatically synchronized with upstream LLVM / LLD / Clang every 4 hours, so
 they are very close to the latest codes upstream.
 
```
@aaronenyeshi an paragraph explaining the interaction between 2 different roles in the process would be good:

- one who conducts the upstream sync process: getting all the codes, merges, file the PRs
- one who reviews the sync PRs

aaronenyeshi committed:
Update SYNC_UPSTREAM.md

Updated the SYNC_UPSTREAM.md as per requested in PR. Added a generic convention for merge branch naming with dates, and 3 paragraphs explaining the roles of merge maintainer with reviewer and their interactions.

aaronenyeshi commented:
I've updated the document as per requested. Let me know if further changes are required here
